src/logics/SubstLoopBreaker.cc

```diff
@@ -205,37 +205,41 @@ std::vector<vec<SNRef>> SubstLoopBreaker::findLoops(vec<SNRef>& startNodes) {
     return loops;
 }
 
-MapWithKeys<PTRef,PtAsgn,PTRefHash> SubstLoopBreaker::constructLooplessSubstitution(MapWithKeys<PTRef,PtAsgn,PTRefHash>&& substs)
+Logic::SubstMap SubstLoopBreaker::constructLooplessSubstitution(MapWithKeys<PTRef,PtAsgn,PTRefHash>&& substs)
 {
+    Logic::SubstMap substPTRefs;
     for (PTRef key : substs.getKeys()) {
         PtAsgn & data = substs[key];
         if (data.sgn != l_True) {
             continue;
         }
 
         SNRef subst_node = sna.getSNRefBySource(key);
-        if (!sna[subst_node].hasChildren())
+        if (!sna[subst_node].hasChildren()) {
             data.sgn = l_False;
```

Comment:
Does this branch serve any purpose anymore?

Reply:
I believe that we can take it off and leave just the else branch.